### PR TITLE
rootfs: Support agent tracing

### LINKED ISSUE #199: Support agent tracing
Allow agent tracing to be added to an image.

Depends on: https://github.com/kata-containers/agent/pull/415.

### PR DESCRIPTION
**Pull request by jodh-intel**
Add `AGENT_TRACE=yes` option to build a rootfs with agent tracing support.

Fixes #199.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: rootfs-builder/rootfs.sh
Comment (grahamwhaley) [inline rootfs-builder/rootfs.sh]: single whitespace indent anomoly?

**Review by grahamwhaley — APPROVED** [review]
changes look good to me.
nice cleanup pickups as well.
lgtm

**Comment by grahamwhaley** [comment]
but you tripped over your own travis URL check :-)
```
INFO: Checking all document URLs
ERROR: Invalid URL 'https://github.com/kata-containers/agent/blob/master/TRACING.md' found in the following files:
```

File: rootfs-builder/rootfs.sh
Comment (jodh-intel) [reply rootfs-builder/rootfs.sh]: Fixed.

**Comment by jodh-intel** [comment]
@grahamwhaley - yep, that was expected - see blocked comment above ;)

**Comment by jodh-intel** [comment]
Added dnm to make it clearer there is a block on the agent PR.

**Comment by jodh-intel** [comment]
This PR is still blocked on https://github.com/kata-containers/agent/pull/415.

**Comment by jodh-intel** [comment]
@sboeuf, @egernst - ptal.

File: rootfs-builder/rootfs.sh
Comment (ydjainopensource) [inline rootfs-builder/rootfs.sh]: Why aren't we checking for this in the script?

**Review by ghost — COMMENTED** [review]
a minor comment lgtm otherwise

File: rootfs-builder/rootfs.sh
Comment (amshinde) [reply rootfs-builder/rootfs.sh]: Agree, would be good to add a check to verify that conflicting values have not been set.

**Comment by jodh-intel** [comment]
Branch updated for latest agent tracing branch (https://github.com/kata-containers/agent/pull/415).

File: rootfs-builder/rootfs.sh
Comment (mcastelino) [inline rootfs-builder/rootfs.sh]: @jodh-intel where is the jaeger-client-socat-redirector.service file? Is is already present in the rootfs?

**Comment by jodh-intel** [comment]
Please do **NOT** remove the `do-not-merge` label as this PR should not land atm.

If it does land, it will currently break agent output since `kata-redirect-agent-output-to-journal.conf` will send all output to the journal, which for a default (proxy) setup is the same as redirecting the output to `/dev/null`.

See: https://github.com/kata-containers/agent/issues/484.

**Comment by egernst** [comment]
@jodh-intel What's the relevance/status of this one?

**Comment by jodh-intel** [comment]
@egernst - it's sitting here "semi-blocked" on https://github.com/kata-containers/runtime/issues/1352; we *could* land this PR now, but as we're about to rework the agent changes for opencensus, there may be no point as this PR won't work with that. I'll leave it up to @mcastelino to decide though.

Related: https://github.com/kata-containers/runtime/issues/1369.

**Comment by jodh-intel** [comment]
https://github.com/kata-containers/agent/pull/415 has now landed so this can now land.

/retest

**Comment by jodh-intel** [comment]
Regarding my comment here: https://github.com/kata-containers/osbuilder/pull/200#issuecomment-473384786, I've reworked the PR slightly so that by default the agent output will *not* be redirected to the journal. Now that https://github.com/kata-containers/shim/pull/172 has landed, this is the behaviour we want I think.

Note also: https://github.com/kata-containers/agent/pull/547.

**Review by devimc — APPROVED** [review]

File: rootfs-builder/alpine/config.sh
Comment (amshinde) [inline rootfs-builder/alpine/config.sh]: @jodh-intel Does this need to go to the Dockerfiles as well? I am not clear if the Dockerfiles are used when USE_DOCKER is true.

File: rootfs-builder/rootfs.sh
Comment (jodh-intel) [reply rootfs-builder/rootfs.sh]: Good catch - fixed.

File: rootfs-builder/alpine/config.sh
Comment (jodh-intel) [reply rootfs-builder/alpine/config.sh]: Not really as the docker files are mostly static - we want `socat` to be added conditionally to avoid bloat at this stage.

**Comment by amshinde** [comment]
@jodh-intel Can you take a look at the CI failures? Will be good to land this PR.
I am going to rerun the CIs , in case they have gone stale.

**Review by egernst — APPROVED** [review]

**Review by egernst — APPROVED** [review]
Looks fine.  AFAICT we don't have any tests in place which will exercise this, correct?

**Comment by egernst** [comment]
CI seems miserable with this change.  @jcvenegas , @chavafg - seems there may be some issues w/ the osbuilder CI?

**Comment by jodh-intel** [comment]
I should probably tal at this PR again tomorrow as it hasn't been touched in a while, so adding dnm for now...

**Comment by fidencio** [comment]
Let me close this one as it's a do-not-merge which won't make to the 1.13.0 release.

**Pull request closed without merging**
